dependencies/aws-sdk-cpp/meson.build

```diff
@@ -0,0 +1,48 @@
+# aws s3 dependency file
+#============================================================================
+
+
+# Look for aws-cpp-sdk, an optional dependency
+#--------------------------------------------------
+if get_option('with_s3')
```

Comment:
Feels like this should not be tested here...
The compiler and flags could be declared regardless of the option being present or not.
I would rather see the option checking happen when declaring the dependencies of `libstore`, to see if this lib should be added or not.